03week/towersOfHanoi.js

```diff
@@ -19,26 +47,51 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+const movePiece=(startStack,endStack)=> {
+ 
+  // Pop removes an item from the startStack and 
+  // the push on the endstack will take in the popped value 
+  // and apply it to endStack. 
+  console.log(stacks[startStack].slice(-1)[0],'hey');
+  return stacks[endStack].push(stacks[startStack].pop());
+  
 }
 
-function isLegal() {
-  // Your code here
 
+const isLegal=(startStack,endStack)=> {
+// the move will be legal if the endstack is an empty array with no values 
+// applied to it 
+// if start stack C has 3 on it, and endStack B is 2,  
+  return stacks[endStack].push() === 0 || stacks[endStack].slice(-1)[0] > stacks[startStack].slice(-1)[0];
 }
 
-function checkForWin() {
-  // Your code here
-
+const checkForWin=()=> {
+ // if the length  is equal to 4 that is a valid win
+  return stacks.c.length === 4 || stacks.b.length === 4;
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
 
+
+const towersOfHanoi=(startStack, endStack)=>{
+  // if legal, then call the move piece function 
+  // if it's NOT legal, than say "invalid play"
+  if (isLegal(startStack, endStack)){
+    movePiece(startStack, endStack)
+  } else {
+    console.log("invalid play");
+  }
+  if (checkForWin()) {
```

Comment:
you should only check for win if the move is legal, here you're checking every time